akka-remote/src/main/scala/akka/remote/BoundAddressesExtension.scala

```diff
@@ -26,7 +27,9 @@ class BoundAddressesExtension(val system: ExtendedActorSystem) extends Extension
   /**
    * Returns a mapping from a protocol to a set of bound addresses.
    */
-  def boundAddresses: Map[String, Set[Address]] = system.provider
-    .asInstanceOf[RemoteActorRefProvider].transport
-    .asInstanceOf[Remoting].boundAddresses
+  def boundAddresses: Map[String, Set[Address]] =
+    system.provider.asInstanceOf[RemoteActorRefProvider].transport match {
+      case r: Remoting        ⇒ r.boundAddresses
+      case a: ArteryTransport ⇒ Map(ArteryTransport.ProtocolName → a.addresses)
+    }
```

Comment:
Does it really make sense to port this, from what I can see it is only used in the `NettyTransportSpec`, and we don't have multiple protocol names any more, so the API feels a bit specific to the old remoting?